Makefile

```diff
@@ -51,35 +51,39 @@ GO_DEPS_LIST = github.com/gorilla/mux \
                github.com/antchfx/xmlquery \
                github.com/facette/natsort \
 	           github.com/philopon/go-toposort \
-			   gopkg.in/godbus/dbus.v5
+	           gopkg.in/godbus/dbus.v5
 
 
 REST_BIN = $(BUILD_DIR)/rest_server/main
 CERTGEN_BIN = $(BUILD_DIR)/rest_server/generate_cert
 
+go-deps = $(BUILD_DIR)/gopkgs/.done
+go-patch = $(BUILD_DIR)/gopkgs/.patch_done
+go-redis-patch = $(BUILD_DIR)/gopkgs/.redis_patch_done
 
-all: build-deps go-deps go-redis-patch go-patch translib rest-server cli
+all: build-deps $(go-deps) $(go-redis-patch) $(go-patch) translib rest-server cli
 
 build-deps:
-	mkdir -p $(BUILD_DIR)
+	mkdir -p $(BUILD_DIR)/gopkgs
 
-go-deps: $(GO_DEPS_LIST)
+$(BUILD_DIR)/gopkgs/.done:
+	$(GO) get -v $(GO_DEPS_LIST)
+	touch  $@
```

Comment:
You need to add $(MAKEFILE_LIST) as prerequisite to such targets. Otherwise the target will never be rerun during incremental builds unless sentinel file is removed. Example, if someone adds a new package name to GO_DEPS_LIST and runs "make go-deps", the new package will not be downloaded. If this was the reason for moving gopkgs directory inside build directory, please revert that change.

Reply:
Thanks. Added the dependency.